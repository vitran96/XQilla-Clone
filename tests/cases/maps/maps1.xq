xquery version "3.0";

declare option xqilla:extensions "true";

for $a in (

let $map := empty-map()
let $map := map-put($map, "a", "aardvark")
let $map := map-put($map, "z", "zebra")
let $map := map-put($map, "e", ("elephant", "eagle"))
let $map := map-put($map, "o", "osterich")
let $map2 := map-put($map, "a", "antelope")
let $map := map-put($map, "t", "terrapin")
let $map3 := map-put($map, 5, "lala")
let $map4 := map-put($map2, "d", 1)
return (
  $map("a"),
  $map2("a"),
  map-get($map, "e"),
  map-contains($map, "k"),
  map-contains($map, "t"),
  map-contains($map2, "t"),

  map-entries($map),
  map-keys($map2),

  map-fold(
    function($a, $k, $v) {
      $a, concat("key: ", $k, ", value: (",
        string-join($v, ", "), ")")
    }, (), $map2),

  $map instance of map(),
  $map instance of map(xs:string, xs:string+),
  $map2 instance of map(xs:string, xs:string),
  $map3 instance of map(xs:string, xs:string*),
  $map3 instance of map(xs:anyAtomicType, xs:string*),
  $map4 instance of map(xs:string, xs:string*),
  $map4 instance of map(xs:string, item()*)
),
let $map := tuple { a := 5, b := 10 }
return (
$map(xs:QName("b")),
map-contains($map, xs:QName("a")),
$map instance of tuple(a as xs:integer),
$map instance of tuple(a as xs:string),
$map instance of tuple(a as xs:decimal, d as xs:date?),
map-new(($map, tuple { c := 99, a := "moo" }))
)

) return (<result>{$a}</result>, "&#10;")
