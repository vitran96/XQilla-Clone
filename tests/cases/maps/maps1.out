<result>aardvark</result>
<result>antelope</result>
<result>elephant</result>
<result>eagle</result>
<result>false</result>
<result>true</result>
<result>false</result>
<result>tuple{key: a, value: aardvark}</result>
<result>tuple{key: z, value: zebra}</result>
<result>tuple{key: e, value: (elephant, eagle)}</result>
<result>tuple{key: o, value: osterich}</result>
<result>tuple{key: t, value: terrapin}</result>
<result>z</result>
<result>e</result>
<result>o</result>
<result>a</result>
<result>key: z, value: (zebra)</result>
<result>key: e, value: (elephant, eagle)</result>
<result>key: o, value: (osterich)</result>
<result>key: a, value: (antelope)</result>
<result>true</result>
<result>true</result>
<result>false</result>
<result>false</result>
<result>true</result>
<result>false</result>
<result>true</result>
<result>10</result>
<result>true</result>
<result>true</result>
<result>false</result>
<result>true</result>
<result>tuple{b: 10, c: 99, a: moo}</result>
